ntbs-service/Models/ModelBase.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using ntbs_service.Models.Enums;
+
+namespace ntbs_service.Models
+{
+    public class ModelBase
+    {
+        [NotMapped]
+        public bool ShouldValidateFull { get; set; }
+
+        public void SetFullValidation(NotificationStatus notificationStatus, bool isBeingSubmitted) 

```

Comment:
I can't access Notification Status from here unless I have back-reference to Notification and then Set it when I am settings property of Notification. This may cause circular references. This is the simplest solution I could think of. open to any other suggestions